embedded-database-spring-test/src/main/java/io/zonky/test/db/EmbeddedDatabaseTestExecutionListener.java

```diff
@@ -87,6 +87,10 @@ private void captureTestClassPreparers(TestContext testContext, RefreshMode... r
     private void forEachDatabase(TestContext testContext, RefreshMode[] refreshModes, BiConsumer<DatabaseContext, AutoConfigureEmbeddedDatabase> action) {
         Set<AutoConfigureEmbeddedDatabase> annotations = AnnotationUtils.getDatabaseAnnotations(testContext.getTestClass());
 
+        if (annotations.isEmpty()) {
```

Comment:
I'm not sure if this brings any difference, because it is only used in a `.forEach()` part. A `.forEach()` on a empty Set will already do nothing.

Reply:
The real issue was in calling the `testContext.getApplicationContext()` method below. This method call could trigger eager initialization of Spring context, even without presence of the `@AutoConfigureEmbeddedDatabase` annotation.